rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -878,10 +878,17 @@ rmw_wait(
 
   // invoke wait until one of the conditions triggers
   DDS::ConditionSeq active_conditions;
+  bool non_blocking = false;
   DDS::Duration_t timeout;
-  timeout.sec = non_blocking ? 0 : 1;
-  timeout.nanosec = 0;
+  if (!wait_timeout) {
+    timeout = DDS::DURATION_INFINITE;
+  } else {
+    non_blocking = wait_timeout->sec == 0 && wait_timeout->nsec == 0;
+    timeout.sec = wait_timeout->sec;
+    timeout.nanosec = wait_timeout->nsec;
+  }
   DDS::ReturnCode_t status = DDS::RETCODE_TIMEOUT;
+
   while (DDS::RETCODE_TIMEOUT == status) {
     status = waitset.wait(active_conditions, timeout);
     if (DDS::RETCODE_TIMEOUT == status) {
```

Comment:
We discussed this off-line, and it seems that the `DURATION_INFINITE` is the right approach, since no additional checks are being done in the busy wait loop anyways. We're relying on our guard condition to wake the waitset up in an exit condition.